.github/workflows/test.yml

```diff
@@ -138,3 +138,24 @@ jobs:
         uses: codecov/codecov-action@v3.1.0
         with:
           fail_ci_if_error: true
+
+  test-nonlinux:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, windows-latest]
+        python-version: [3.7, 3.8]
```

Comment:
Minor nitpick: I think this is slightly more clear if the `runs-on` attribute comes _after_ the strategy matrix as is done in the [github documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-a-multi-dimension-matrix) on this.